### PR TITLE
Remove the heatmap and fix the table tabs to reflect benefits, credits and taxes

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
I will not merge this before adding back the heatmap - using the PR to get intermediate feedback for the format

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,261 +1,292 @@
...
+    "This application evaluates marriage penalties and bonuses of couples, based on state and individual employment income"
+)
+st.markdown(
+    "This application utilizes <a href='https://github.com/PolicyEngine/us-marriage-incentive'>the policyengine API</a>",
+    unsafe_allow_html=True,
+)
+
+# Streamlit inputs for state code, head income, and spouse income.
+statecodes = [s.value for s in StateCode]
+us_territories = {
```
Comment (MaxGhenis) [inline app.py:44]: capitalize constants
```suggestion
US_TERRITORIES = {
```

File: app.py [hunk 2]
```diff
@@ -1,261 +1,292 @@
...
-    "Taxes": ["household_tax_before_refundable_credits","employment_income"] , 
-    "Credits": ["household_refundable_tax_credits", "employment_income"]
+
+# Streamlit heading and description
+st.header("Marriage Incentive Calculator")
+st.write(
+    "This application evaluates marriage penalties and bonuses of couples, based on state and individual employment income"
+)
+st.markdown(
+    "This application utilizes <a href='https://github.com/PolicyEngine/us-marriage-incentive'>the policyengine API</a>",
```
Comment (MaxGhenis) [inline app.py:38]: The repo's code is in the upper right of all streamlit apps. Was something not working with the normal markdown link syntax? No longer need `unsafe_allow_html=True`.
```suggestion
    "This application utilizes the [`policyengine-us` Python package](https://github.com/policyengine/policyengine-us).",
```

File: app.py [hunk 3]
```diff
@@ -1,261 +1,292 @@
...
+    )
     simulation = Simulation(situation=situation)
 
-    #benefits breakdown
     benefits_categories = HouseholdBenefits.adds
 
+    taxes_before_refundable_credits = HouseholdTaxBeforeRefundableCredits.adds
+
+    package = "policyengine_us"
+    resource_path_federal = "parameters/gov/irs/credits/refundable.yaml"
```
Comment (MaxGhenis) [inline app.py:128]: the way we do this in the web app is to navigate the `adds` features of the variables: https://github.com/PolicyEngine/policyengine-app/blob/7e3b1b6117de799f7ef19d199138ea96af93e971/src/pages/household/output/NetIncomeBreakdown.jsx#L176

@nikhilwoodruff do you have an example of doing this in Python?

File: app.py [hunk 4]
```diff
@@ -1,261 +1,292 @@
...
-    
-    benefits_dic ={}
+    household_refundable_tax_credits = int(
+        simulation.calculate("household_refundable_tax_credits", int(YEAR))
+    )
+    household_tax_before_refundable_credits = int(
+        simulation.calculate("household_tax_before_refundable_credits", int(YEAR))
+    )
+
+    benefits_dic = {}
```
Comment (MaxGhenis) [inline app.py:147]: by convention
```suggestion
    benefits_dict = {}
```

File: app.py [hunk 5]
```diff
@@ -1,261 +1,292 @@
...
-    # separate programs
-    head_separate = programs[1][:-1] # we exclude the last element which is the dictionary of benefits breakdown 
-    spouse_separate = programs[2][:-1] # we exclude the last element which is the dictionary of benefits breakdown 
-    separate = [x + y for x, y in zip(head_separate, spouse_separate)]
-    formatted_separate = list(map(lambda x: "${:,}".format(round(x)), separate))
-    
-    # delta
-    delta = [x - y for x, y in zip(married_programs, separate)]
-    delta_percent = [(x - y) / x if x != 0 and x != 0 else 0 for x, y in zip(married_programs, separate)]
+    formatted_benefits_married = list(
```
Comment (MaxGhenis) [inline app.py:243]: make this a function since applying 4x

File: app.py [hunk 6]
```diff
@@ -1,261 +1,292 @@
...
+
+    formatted_delta = [
+        "${:,}".format(round(x)) if isinstance(x, (int, float)) else x for x in delta
+    ]
+    formatted_delta_percent = [
+        "{:.1%}".format(x) if isinstance(x, (int, float)) else x for x in delta_percent
+    ]
+
+    programs_list = [
+        "Net Income",
```
Comment (MaxGhenis) [inline app.py:282]:
```suggestion
        "Net income",
```

File: app.py [hunk 7]
```diff
@@ -264,245 +295,133 @@ def summarize_marriage_bonus(marriage_bonus):
...
-            scaleratio=1,
-        )
-    )
-   
-    fig.update_layout(height=600, width=800)
-    
-    # Customize hover template to display x and y values along with the percentage change
-    fig.update_traces(hovertemplate='Head Employment Income: %{x}<br>Spouse Employment Income: %{y}<br>Change: %{customdata:.2f}%')
 
+    taxes_before_refundable_credits = list(programs[0][-3].keys())
```
Comment (MaxGhenis) [inline app.py:383]: you can avoid code duplication by creating a single dataframe with all the variables you want across the three situations, and adding a column for the category (Total, Taxes before refundable credits, Refundable credits, Benefits) then when displaying the breakdown filtering the table on that category. Then you can calculate absolute and relative differences in one dataframe command.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
also run black for formatting

**Comment by PavelMakarchuk** [comment]
> I will not merge this before adding back the heatmap - using the PR to get intermediate feedback for the format

Decided to merge to match the PR and limit PR complexity for review -> Will add heatmap in separate PR

**Pull request merged**